### PR TITLE
[20129] Type Object integration

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
## Description

This PR integrates the new TypeObject support feature.

## Contributor Checklist
- [ ] Commit messages follow the project guidelines. 
- [ ] The code follows the style guidelines of this project. 
- [ ] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [ ] Any new/modified methods have been properly documented using Doxygen. 
- [ ] Changes are ABI compatible. 
- [ ] Changes are API compatible. 
- [ ] New feature has been added to the `versions.md` file (if applicable).
- [ ] New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [ ] Applicable backports have been included in the description.

## Reviewer Checklist
- [ ] The PR has a milestone assigned.
- [ ] The title and description correctly express the PR's purpose.
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Review by richiware — CHANGES_REQUESTED** [review]

File: include/fastdds/dds/xtypes/type_representation/detail/dds_xtypes_typeobjectPubSubTypes.h [hunk 1]
```diff
@@ -215,6 +212,9 @@ class StringSTypeDefnPubSubType : public eprosima::fastdds::dds::TopicDataType
     eProsima_user_DllExport void deleteData(
             void* data) override;
 
+    //Register TypeObject representation in Fast DDS TypeObjectRegistry
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply include/fastdds/dds/xtypes/type_representation/detail/dds_xtypes_typeobjectPubSubTypes.h:215]: The method implementation has a warning stating that there is no support for the TypeObject representation as it was disabled in the generated code.

File: src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterFactory.cpp [hunk 2]
```diff
@@ -517,19 +516,53 @@ ReturnCode_t DDSFilterFactory::create_content_filter(
...
         else
         {
-            auto node = parser::parse_filter_expression(filter_expression, type_object);
+            auto node = parser::parse_filter_expression(filter_expression, &type_object);
             if (node)
             {
-                auto type_id = TypeObjectFactory::get_instance()->get_type_identifier(type_name, true);
-                auto dyn_type = TypeObjectFactory::get_instance()->build_dynamic_type(type_name, type_id, type_object);
+                eprosima::fastdds::dds::xtypes::TypeIdentifierPair type_ids;
+                eprosima::fastdds::dds::xtypes::TypeIdentifier type_id;
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterFactory.cpp:534]: Good catch. The TypeIdentifier does not seem to be needed anymore.

File: src/cpp/fastrtps_deprecated/Domain.cpp [hunk 3]
```diff
@@ -350,47 +348,47 @@ bool Domain::registerType(
...
-
-            // Complete, just to make sure it is generated
-            const TypeIdentifier* type_id_complete = typeFactory->get_type_identifier(type->getName(), true);
-            const TypeObject* type_obj_complete = typeFactory->get_type_object(type->getName(), true);
-            typeFactory->add_type_object(type->getName(), type_id_complete, type_obj_complete); // Add complete
-        }
-    }
-    return registerType(part, type);
-}
+// bool Domain::registerDynamicType(
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply src/cpp/fastrtps_deprecated/Domain.cpp:351]: This file is going to be removed in Fast DDS 3.0.0 so nothing is required to be done.

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

File: test/blackbox/api/dds-pim/PubSubReader.hpp [hunk 4]
```diff
@@ -27,32 +27,32 @@
...
-#include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
 #include <fastdds/dds/topic/Topic.hpp>
 #include <fastdds/rtps/transport/UDPTransportDescriptor.h>
 #include <fastdds/rtps/transport/UDPv4TransportDescriptor.h>
 #include <fastdds/rtps/transport/UDPv6TransportDescriptor.h>
+#include <fastrtps/utils/IPLocator.h>
 #include <fastrtps/xmlparser/XMLParser.h>
 #include <fastrtps/xmlparser/XMLTree.h>
-#include <fastrtps/utils/IPLocator.h>
+#include <gtest/gtest.h>
```
Comment (richiware) [inline test/blackbox/api/dds-pim/PubSubReader.hpp:52]: Why do you move external includes at the end, not according to our coding style?

**Review by richiware — CHANGES_REQUESTED** [review]

**Review by richiware — DISMISSED** [review]
Ready when CI ready

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Pull request merged**